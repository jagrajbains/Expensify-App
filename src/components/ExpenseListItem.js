import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";
import "../locales/inr";
const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  numeral.locale("inr");
  return (
    <div>
      <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
      </Link>
      <p>
        {numeral(amount).format("$0,0.00")}-
        {moment(createdAt).format("ddd, MMMM Do YYYY")}
      </p>
    </div>
  );
};

export default ExpenseListItem;
