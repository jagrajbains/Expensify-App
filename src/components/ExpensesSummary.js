import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import "../locales/inr";
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expense-total";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  numeral.locale("inr");
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(expensesTotal).format("$0,0.00");
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = state => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
