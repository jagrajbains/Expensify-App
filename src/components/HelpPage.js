import React from 'react';

const HelpPage = () => (
    <div>
        <p>This is from the help Page</p>
    </div>
);

export default HelpPage;