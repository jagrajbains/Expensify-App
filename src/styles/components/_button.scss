.button {
  background-color: $blue;
  border-radius: 4px;
  border: none;
  color: #fff;
  display: inline-block;
  font-weight: 300;
  line-height: 1;
  padding: $s-size $m-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--danger {
  @extend .button;
  background-color: #d9534f;
}
