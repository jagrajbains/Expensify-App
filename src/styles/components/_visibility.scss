.show-for-mobile {
  @media (min-width: $desktop-breakpoint) {
    display: none;
  }
}

.show-for-largerScreens {
  @media (max-width: $desktop-breakpoint - 0.01rem) {
    display: none;
  }
}
