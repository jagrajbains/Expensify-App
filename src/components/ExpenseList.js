import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import selectedExpenses from "../selectors/expenses";

export const ExpenseList = props => {
  return (
    <div className="content-container">
      <div className="list-header">
        <div className="show-for-mobile">Expenses</div>
        <div className="show-for-largerScreens">Expense</div>
        <div className="show-for-largerScreens">Amount</div>
      </div>
      <div className="list-body">
        {props.expenses.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No Expenses Added.</span>
          </div>
        ) : (
          props.expenses.map(expense => {
            return <ExpenseListItem key={expense.id} {...expense} />;
          })
        )}
      </div>
    </div>
  );
};

const mapStateToProps = state => {
  return {
    expenses: selectedExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
