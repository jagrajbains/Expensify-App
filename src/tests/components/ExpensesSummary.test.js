import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("Should render ExpensesSummary with 1 expense correctly.", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={245.67} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("Should render ExpensesSummary with multiple expenses correctly.", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={23} expensesTotal={25478.23546} />
  );
  expect(wrapper).toMatchSnapshot();
});
