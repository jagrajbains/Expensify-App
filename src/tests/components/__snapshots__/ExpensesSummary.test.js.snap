// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense correctly. 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    ₹245.67
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with multiple expenses correctly. 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    ₹25,478.24
  </h1>
</div>
`;
